File: sdk/php/tests/integration_test.php
<?php

require_once __DIR__ . '/../src/Database.php';
require_once __DIR__ . '/../src/Filter.php';

use LibraVDB\Database;
use LibraVDB\Filter;

$dbPath = "./test_db_php_" . getmypid();

// 1. Open Database
echo "Opening DB...\n";
$db = new Database($dbPath);
if (!$db) {
    die("Failed to open DB\n");
}

// 2. Create Collection
echo "Creating Collection...\n";
$col = $db->createCollection("test_col", 3);
if (!$col) {
    die("Failed to create collection\n");
}

// 3. Insert and Search
echo "Inserting...\n";
$col->insert("1", [1.0, 2.0, 3.0], '{"category": "A"}');

$filter = Filter::eq("category", "A");
$filterJson = Filter::toJson($filter);

echo "Searching...\n";
$searchRes = $col->search([1.0, 2.0, 3.0], 10, $filterJson);
if (!str_contains($searchRes, '"id":"1"')) {
    die("Search results missing inserted vector\n");
}

// 4. Batch Insert
echo "Batch Inserting...\n";
$col->insertBatch(
    ["2", "3"],
    [
        [4.0, 5.0, 6.0],
        [7.0, 8.0, 9.0]
    ],
    ['{"category": "B"}', '{"category": "C"}']
);

// 5. Scan
echo "Scanning...\n";
$scanRes = $col->scan();
if (!str_contains($scanRes, '"id":"2"') || !str_contains($scanRes, '"id":"3"')) {
    die("Scan results missing batch inserted vectors\n");
}

// Cleanup
$db->close();
system("rm -rf " . escapeshellarg($dbPath));

echo "All PHP integration tests passed!\n";

File: sdk/csharp/LibraVDB.Tests/IntegrationTest.cs
using System;
using System.IO;
using System.Text.Json;
using Xunit;
using LibraVDB;

namespace LibraVDB.Tests
{
    public class IntegrationTest : IDisposable
    {
        private readonly string _dbPath = $"./test_db_{Guid.NewGuid()}";

        public IntegrationTest()
        {
            if (Directory.Exists(_dbPath))
            {
                Directory.Delete(_dbPath, true);
            }
        }

        public void Dispose()
        {
            if (Directory.Exists(_dbPath))
            {
                Directory.Delete(_dbPath, true);
            }
        }

        [Fact]
        public void TestEndToEnd()
        {
            using (var db = new Database(_dbPath))
            {
                var col = db.CreateCollection("test_col", 3);

                // Insert
                col.Insert("1", new float[] { 1.0f, 2.0f, 3.0f }, new { category = "A" });

                // Query
                string results = col.Search(new float[] { 1.0f, 2.0f, 3.0f }, 10, Filter.Eq("category", "A"));
                Assert.Contains("\"id\":\"1\"", results);

                // Insert Batch
                string[] batchIds = new string[] { "2", "3" };
                float[][] batchVectors = new float[][] {
                    new float[] { 4.0f, 5.0f, 6.0f },
                    new float[] { 7.0f, 8.0f, 9.0f }
                };
                object[] batchMeta = new object[] {
                    new { category = "B" },
                    new { category = "C" }
                };

                col.InsertBatch(batchIds, batchVectors, batchMeta);

                // Scan
                string scanRes = col.Scan();
                Assert.Contains("\"id\":\"1\"", scanRes);
                Assert.Contains("\"id\":\"2\"", scanRes);
                Assert.Contains("\"id\":\"3\"", scanRes);

                // Delete
                col.Delete("2");
                scanRes = col.Scan();
                Assert.DoesNotContain("\"id\":\"2\"", scanRes);

                // Vacuum
                db.Vacuum();
            }
        }
    }
}
